### PR TITLE
Expose PreAuthResponse item initializer

### PR DESCRIPTION
**Pull request by GhenadieVP**
Expose the uniffi constructor method to create the `WalletToDappInteractionPreAuthorizationResponseItems` from `Subintent` and `signatures`.

### CODE REVIEW

File: crates/sargon/src/signing/signables/signable_subintent.rs [hunk 1]
```diff
@@ -33,7 +33,9 @@ impl Signable for Subintent {
             );
         }
 
-        let manifest = TransactionManifestV2::sargon_built(builder, network_id);
+        builder = builder.yield_to_parent(());
```
Comment (GhenadieVP) [inline crates/sargon/src/signing/signables/signable_subintent.rs:36]: SubintentManifest should always have at least `yield_to_parent` instruction to be valid.

File: crates/sargon/src/wrapped_radix_engine_toolkit/low_level/v2/subintent.rs [hunk 2]
```diff
@@ -3,20 +3,24 @@ use delegate::delegate;
 use std::hash::Hasher;
 
 #[derive(Clone, PartialEq, Eq, derive_more::Debug)]
-#[debug("{:?}", self.intent_core)]
+#[debug("header:\n{:?}\n\nmessage:\n{:?}\n\nmanifest:\n{}\n\n", self.header, self.message, self.manifest.manifest_string())]
 pub struct Subintent {
-    pub intent_core: IntentCoreV2,
+    pub header: IntentHeaderV2,
```
Comment (GhenadieVP) [inline crates/sargon/src/wrapped_radix_engine_toolkit/low_level/v2/subintent.rs:8]: Subintent was incorrectly represented through TransactionManifestV2, it should be SubintentManifest.

File: crates/sargon/src/radix_connect/wallet_interaction/dapp_wallet_interaction/wallet_to_dapp/success_response/pre_authorization/pre_authorization.rs [hunk 3]
```diff
@@ -32,13 +32,20 @@ impl WalletToDappInteractionPreAuthorizationResponseItems {
...
+    pub fn new_with_subintent_and_signatures(
+        subintent: Subintent,
+        signatures: Vec<IntentSignature>,
+    ) -> Result<Self> {
+        Self::new(build_signed_partial_transaction(subintent, signatures))
+    }
 }
 
 impl HasSampleValues for WalletToDappInteractionPreAuthorizationResponseItems {
     fn sample() -> Self {
```
Comment (CyonAlexRDX) [inline crates/sargon/src/radix_connect/wallet_interaction/dapp_wallet_interaction/wallet_to_dapp/success_response/pre_authorization/pre_authorization.rs:45]: any value for `sample_other`? :)

**Review by CyonAlexRDX — APPROVED** [review]
Ah, I had missed, so we have `ManifestV1`, `ManifestV2` and `SubIntentManifest` types in RET?

**Comment by GhenadieVP** [comment]
> Ah, I had missed, so we have `ManifestV1`, `ManifestV2` and `SubIntentManifest` types in RET?

Yes, more like in scrypto itself.

File: crates/sargon/src/radix_connect/wallet_interaction/dapp_wallet_interaction/wallet_to_dapp/success_response/pre_authorization/pre_authorization.rs (hunk 3 above)
Comment (GhenadieVP) [reply crates/sargon/src/radix_connect/wallet_interaction/dapp_wallet_interaction/wallet_to_dapp/success_response/pre_authorization/pre_authorization.rs:45]: will add

**Review by matiasbzurovski — APPROVED** [review]
LGTM, only one question

Also, to avoid doing a new release just for this, can you please add a new Swift class with the following content:
```Swift
// CompiledSubintent+Wrap+Functions.swift
extension CompiledSubintent {
	public var data: Data {
		compiledSubintentBytes(compiledIntent: self)
	}
}
```

File: crates/sargon/src/wrapped_radix_engine_toolkit/low_level/v2/subintent.rs (hunk 2 above)
Comment (GhenadieVP) [reply crates/sargon/src/wrapped_radix_engine_toolkit/low_level/v2/subintent.rs:8]: nope, removed.

File: crates/sargon-uniffi/src/wrapped_radix_engine_toolkit/low_level/v2/subintent.rs [hunk 4]
```diff
@@ -3,7 +3,19 @@ use sargon::Subintent as InternalSubintent;
 
 #[derive(Clone, PartialEq, Eq, InternalConversion, uniffi::Record)]
 pub struct Subintent {
-    pub intent_core: IntentCoreV2,
+    pub header: IntentHeaderV2,
+    pub manifest: SubintentManifest,
+    pub message: MessageV2,
```
Comment (micbakos-rdx) [inline crates/sargon-uniffi/src/wrapped_radix_engine_toolkit/low_level/v2/subintent.rs:8]: better like this

**Review by micbakos-rdx — APPROVED** [review]
Thank you for adding kotlin extensions

**Pull request merged**